### PR TITLE
Clean provisioners

### PR DESCRIPTION
**Pull request by SwampDragons**
Refactor provisioner acceptance tests to shell directly out to Packer rather than import the core. Modify test case formatting to more closely reflect the builder test cases.

### CODE REVIEW

File: packer-plugin-sdk/acctest/provisioneracc/provisioners.go [hunk 1]
```diff
@@ -3,91 +3,252 @@ package provisioneracc
...
-					filePath,
+				fmt.Fprintf(out, `{"builders": [%s],"provisioners": [%s]}`,
+					builderFragment, testCase.Template)
+				templateName := fmt.Sprintf("%s_%s.json", builderType, testCase.Type)
+				templatePath := filepath.Join("./", templateName)
+				writeJsonTemplate(out, templatePath, t)
+				logfile := fmt.Sprintf("packer_log_%s_%s.txt", builderType, testCase.Type)
+
+				// Run build
+				buildCommand := exec.Command("packer", "build", "--machine-readable", templatePath)
```
Comment (sylviamoss) [inline packer-plugin-sdk/acctest/provisioneracc/provisioners.go:210]: nice!!

File: packer-plugin-sdk/acctest/provisioneracc/provisioners.go [hunk 2]
```diff
@@ -119,32 +280,52 @@ func writeJsonTemplate(out *bytes.Buffer, filePath string, t *testing.T) {
...
-	GetName() string
-	GetConfig() (string, error)
-	GetProvisionerStore() packersdk.MapOfProvisioner
-	IsCompatible(builder string, vmOS string) bool
-	RunTest(c *command.BuildCommand, args []string) error
-}
-
+// BuilderAcceptance is specialized tooling implemented by individual builders
+// To add your builder to the provisioner testing framework, create a struct
+// that implements the this interface, add it to the BuildersAccTest map below.
```
Comment (sylviamoss) [inline packer-plugin-sdk/acctest/provisioneracc/provisioners.go:285]:
```suggestion
// that implements this interface, add it to the BuildersAccTest map below.
```

File: provisioner/powershell/provisioner_acc_test.go [hunk 3]
```diff
@@ -1,94 +1,110 @@
...
 
-func (s *PowershellProvisionerAccTest) GetName() string {
-	return TestProvisionerName
+	return string(fragmentString), nil
 }
 
-func (s *PowershellProvisionerAccTest) GetConfig() (string, error) {
-	filePath := filepath.Join("./test-fixtures", s.ConfigName)
-	config, err := os.Open(filePath)
+func TestAccPowershellProvisioner_basic(t *testing.T) {
```
Comment (sylviamoss) [inline provisioner/powershell/provisioner_acc_test.go:43]: I like the way it looks now!

**Review by sylviamoss — APPROVED** [review]
Nice done, LGTM!
In the near future, we need to figure out how to make this work for hcl2, or maybe implement something similar.

File: packer-plugin-sdk/acctest/provisioneracc/provisioners.go (hunk 2 above)
Comment (SwampDragons) [reply packer-plugin-sdk/acctest/provisioneracc/provisioners.go:285]: good catch!

File: provisioner/powershell/provisioner_acc_test.go (hunk 3 above)
Comment (SwampDragons) [reply provisioner/powershell/provisioner_acc_test.go:43]: thanks! I kind of copied the builder testcase which means the builder and provisioner acceptance tests now kinda look the same and I think that's nice for us as maintainers.

File: packer-plugin-sdk/acctest/provisioneracc/provisioners.go (hunk 1 above)
Comment (nywilken) [reply packer-plugin-sdk/acctest/provisioneracc/provisioners.go:210]: If a user does not have Packer installed locally the command will error not so nicely.

```
⇶  ACC_TEST_BUILDERS=amazon-ebs make provisioners-acctest TEST=./provisioner/shell
ACC_TEST_BUILDERS=amazon-ebs go test ./provisioner/shell  -timeout=1h
--- FAIL: TestAccShellProvisioner_basic (0.01s)
    --- FAIL: TestAccShellProvisioner_basic/shell-provisioner-basic_on_Amazon-ebs_Linux_builder (0.01s)
        provisioners.go:237: Error running provisioner acceptance tests: %!s(MISSING)
            Logs can be found at %!s(MISSING) and the acceptance test template can be found at %!s(MISSING)Unable find /home/wilken/Development/pr-reviews/packer/provisioner/shell/test-fixtures/provisioner.shell.txtpacker_log_amazon-ebs_shell.txtamazon-ebs_shell.json
FAIL
FAIL    github.com/hashicorp/packer/provisioner/shell   0.094s
FAIL
make: *** [Makefile:144: provisioners-acctest] Error 1

``` 
We should add an `exec.LookPath("packer")` check here with a detailed error message.

**Review by nywilken — APPROVED** [review]
This looks very good. Really nice approach taken here. 

I left one comment around the error messaging, but otherwise good to go.

File: packer-plugin-sdk/acctest/provisioneracc/provisioners.go (hunk 1 above)
Comment (sylviamoss) [reply packer-plugin-sdk/acctest/provisioneracc/provisioners.go:210]: good cath!

**Comment by SwampDragons** [comment]
You're right. I got lazy and meant to come back to that and didn't 😭 I'll tweak it now

**Review by nywilken — APPROVED** [review]
:+1: 
```
    --- FAIL: TestAccShellProvisioner_basic/shell-provisioner-basic_on_Amazon-ebs_Linux_builder (0.01s)
        provisioners.go:212: Couldn't find packer binary installed on system: exec: "packer": executable file not found in $PATH
```

File: Makefile [hunk 4]
```diff
@@ -140,8 +140,8 @@ test: mode-check vet ## Run unit tests
 	@go test -count $(COUNT) $(TEST) $(TESTARGS) -timeout=3m
 
 # acctest runs provisioners acceptance tests
-provisioners-acctest: install-build-deps generate
-	ACC_TEST_BUILDERS=$(ACC_TEST_BUILDERS) ACC_TEST_PROVISIONERS=$(ACC_TEST_PROVISIONERS) go test ./provisioner/... -timeout=1h
+provisioners-acctest: #install-build-deps generate
```
Comment (azr) [inline Makefile:143]: Q: was this on purpose ? It's part of 76177b50cef6127529b6623b848c9535d3e1a117 but looking at the commit message I am not sure 🙂

Comment (SwampDragons) [reply Makefile:143]: I didn't mean to commit it but honestly it's probably better this way. The generation takes longer than the acceptance tests and it's realllly annoying as the person running the tests.

Comment (azr) [reply Makefile:143]: got it !

**Pull request merged**